File: ccif/src/java/other/org/himalay/ccif/ServiceSocket.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:55 EST 2016")

public  class ServiceSocket extends BinStruct implements PublicBinMsg{




    // members variables
    // protocolFlag
    public short protocolFlag ;
    // localPort
    public int localPort ;
    // remotePort
    public int remotePort ;
    // remoteAddressType
    public short remoteAddressType ;
    // name
    public ByteArrayWrapper_ui8 name ;
    // ipV4Address
    public IPV4Address ipV4Address ;
    // ipV6Address
    public IPV6Address ipV6Address ;

        public ServiceSocket () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize protocolFlag
        
        // Initialize localPort
        
        // Initialize remotePort
        
        // Initialize remoteAddressType
        
        // Initialize name
        //Conditional, will be initialized during read
        // Initialize ipV4Address
        //Conditional, will be initialized during read
        // Initialize ipV6Address
        //Conditional, will be initialized during read
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read protocolFlag
        {protocolFlag=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read localPort
        {localPort=istream.readUnsignedShort(); retVal+=2;}
        // read remotePort
        {remotePort=istream.readUnsignedShort(); retVal+=2;}
        // read remoteAddressType
        {remoteAddressType=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read name
        if (remoteAddressType==0x00){ name = new ByteArrayWrapper_ui8();retVal += name.read(istream);        }
        // read ipV4Address
        if (remoteAddressType==0x01){ ipV4Address = new IPV4Address();retVal += ipV4Address.read(istream);        }
        // read ipV6Address
        if (remoteAddressType==0x02){ ipV6Address = new IPV6Address();retVal += ipV6Address.read(istream);        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        
        
        
        
        { /** fix dependent sizes for name **/  }
        { /** fix dependent sizes for ipV4Address **/  }
        { /** fix dependent sizes for ipV6Address **/  }
    
                // write protocolFlag
        ostream.writeByte(protocolFlag); retVal +=1;
        // write localPort
        ostream.writeShort(localPort); retVal +=2;
        // write remotePort
        ostream.writeShort(remotePort); retVal +=2;
        // write remoteAddressType
        ostream.writeByte(remoteAddressType); retVal +=1;
        // write name
        if (name!=null)retVal +=name.write(ostream);
        // write ipV4Address
        if (ipV4Address!=null)retVal +=ipV4Address.write(ostream);
        // write ipV6Address
        if (ipV6Address!=null)retVal +=ipV6Address.write(ostream);
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("ServiceSocket\n");
    dc.increaseIndent();
        int retVal= 0;
        // write protocolFlag
        dc.indent();dc.getPs().println("protocolFlag="+protocolFlag+"(0x"+ Integer.toHexString(protocolFlag) + ")" );
        // write localPort
        dc.indent();dc.getPs().println("localPort="+localPort+"(0x"+ Integer.toHexString(localPort) + ")" );
        // write remotePort
        dc.indent();dc.getPs().println("remotePort="+remotePort+"(0x"+ Integer.toHexString(remotePort) + ")" );
        // write remoteAddressType
        dc.indent();dc.getPs().println("remoteAddressType="+remoteAddressType+"(0x"+ Integer.toHexString(remoteAddressType) + ")" );
        // write name
        if ( name != null ) {dc.indent();dc.getPs().println("name") ;retVal +=name.dump(dc);}
        // write ipV4Address
        if ( ipV4Address != null ) {dc.indent();dc.getPs().println("ipV4Address") ;retVal +=ipV4Address.dump(dc);}
        // write ipV6Address
        if ( ipV6Address != null ) {dc.indent();dc.getPs().println("ipV6Address") ;retVal +=ipV6Address.dump(dc);}
dc.decreaseIndent();
        return retVal;
    }


        // Getter for protocolFlag
    //public short getProtocolFlag()
    //{
    //    return protocolFlag ;
    //}

    
    // Setter for protocolFlag
    //public void setProtocolFlag(short val)
    //{
    //    this.protocolFlag= val;
    //}
    // Getter for localPort
    //public int getLocalPort()
    //{
    //    return localPort ;
    //}

    
    // Setter for localPort
    //public void setLocalPort(int val)
    //{
    //    this.localPort= val;
    //}
    // Getter for remotePort
    //public int getRemotePort()
    //{
    //    return remotePort ;
    //}

    
    // Setter for remotePort
    //public void setRemotePort(int val)
    //{
    //    this.remotePort= val;
    //}
    // Getter for remoteAddressType
    //public short getRemoteAddressType()
    //{
    //    return remoteAddressType ;
    //}

    
    // Setter for remoteAddressType
    //public void setRemoteAddressType(short val)
    //{
    //    this.remoteAddressType= val;
    //}
    // Getter for name
    //public ByteArrayWrapper_ui8 getName()
    //{
    //    return name ;
    //}

    
    // Setter for name
    //public void setName(ByteArrayWrapper_ui8 val)
    //{
    //    this.name= val;
    //}
    // Getter for ipV4Address
    //public IPV4Address getIpV4Address()
    //{
    //    return ipV4Address ;
    //}

    
    // Setter for ipV4Address
    //public void setIpV4Address(IPV4Address val)
    //{
    //    this.ipV4Address= val;
    //}
    // Getter for ipV6Address
    //public IPV6Address getIpV6Address()
    //{
    //    return ipV6Address ;
    //}

    
    // Setter for ipV6Address
    //public void setIpV6Address(IPV6Address val)
    //{
    //    this.ipV6Address= val;
    //}


    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/ServiceSocket.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:12 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "ServiceSocket.h"

#ifdef org_himalay_ccif_ServiceSocket__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
ServiceSocket::ServiceSocket() {
    references= 0;


    // protocolFlag
    protocolFlag= (ui8)0;
    // localPort
    localPort= (ui16)0;
    // remotePort
    remotePort= (ui16)0;
    // remoteAddressType
    remoteAddressType= (ui8)0;
    // name
    org_himalay_ccif_ByteArrayWrapper_ui8__INIT(name)
    // ipV4Address
    org_himalay_ccif_IPV4Address__INIT(ipV4Address)
    // ipV6Address
    org_himalay_ccif_IPV6Address__INIT(ipV6Address)
}
// Destructor
ServiceSocket::~ServiceSocket() {
    // protocolFlag

    // localPort

    // remotePort

    // remoteAddressType

    // name
    org_himalay_ccif_ByteArrayWrapper_ui8__DELETE( name);
    // ipV4Address
    org_himalay_ccif_IPV4Address__DELETE( ipV4Address);
    // ipV6Address
    org_himalay_ccif_IPV6Address__DELETE( ipV6Address);
}

int ServiceSocket::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // protocolFlag
    {
        protocolFlag=(istream.readUI8());
        retVal+=1;
    }
    // localPort
    {
        localPort=istream.readUI16();
        retVal+=2;
    }
    // remotePort
    {
        remotePort=istream.readUI16();
        retVal+=2;
    }
    // remoteAddressType
    {
        remoteAddressType=(istream.readUI8());
        retVal+=1;
    }
    // name
    if (remoteAddressType==0x00) {
        org_himalay_ccif_ByteArrayWrapper_ui8__NEW( name);
        retVal += name->read(istream);
    }
    // ipV4Address
    if (remoteAddressType==0x01) {
        org_himalay_ccif_IPV4Address__NEW( ipV4Address);
        retVal += ipV4Address->read(istream);
    }
    // ipV6Address
    if (remoteAddressType==0x02) {
        org_himalay_ccif_IPV6Address__NEW( ipV6Address);
        retVal += ipV6Address->read(istream);
    }
    return retVal;
}

int ServiceSocket::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read protocolFlag
    {
        protocolFlag=(istream.readUI8());
        retVal+=1;
    }
    // read localPort
    {
        localPort=istream.readUI16();
        retVal+=2;
    }
    // read remotePort
    {
        remotePort=istream.readUI16();
        retVal+=2;
    }
    // read remoteAddressType
    {
        remoteAddressType=(istream.readUI8());
        retVal+=1;
    }
    // read name
    if (remoteAddressType==0x00) {
        org_himalay_ccif_ByteArrayWrapper_ui8__NEW( name);
        retVal += name->read(istream);
    }
    // read ipV4Address
    if (remoteAddressType==0x01) {
        org_himalay_ccif_IPV4Address__NEW( ipV4Address);
        retVal += ipV4Address->read(istream);
    }
    // read ipV6Address
    if (remoteAddressType==0x02) {
        org_himalay_ccif_IPV6Address__NEW( ipV6Address);
        retVal += ipV6Address->read(istream);
    }

    return retVal;
}

int ServiceSocket::write(DataOutputStream& ostream) { // throws IOException




    {   /** fix dependent sizes for name **/
    }
    {   /** fix dependent sizes for ipV4Address **/
    }
    {   /** fix dependent sizes for ipV6Address **/
    }

    int retVal= 0;
    // write protocolFlag
    ostream.writeUI8(protocolFlag);
    retVal +=1;
    // write localPort
    ostream.writeUI16(localPort);
    retVal +=2;
    // write remotePort
    ostream.writeUI16(remotePort);
    retVal +=2;
    // write remoteAddressType
    ostream.writeUI8(remoteAddressType);
    retVal +=1;
    // write name
    if ( name != NULL ) {
        retVal +=name->write(ostream);
    }
    // write ipV4Address
    if ( ipV4Address != NULL ) {
        retVal +=ipV4Address->write(ostream);
    }
    // write ipV6Address
    if ( ipV6Address != NULL ) {
        retVal +=ipV6Address->write(ostream);
    }

    return retVal;
}

int ServiceSocket::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("ServiceSocket\n");
    dc.increaseIndent();
    int retVal= 0;
    // write protocolFlag
    dc.indent();
    dc.getPs().print("protocolFlag=");
    dc.getPs().println((long)protocolFlag);
    // write localPort
    dc.indent();
    dc.getPs().print("localPort=");
    dc.getPs().println((long)localPort);
    // write remotePort
    dc.indent();
    dc.getPs().print("remotePort=");
    dc.getPs().println((long)remotePort);
    // write remoteAddressType
    dc.indent();
    dc.getPs().print("remoteAddressType=");
    dc.getPs().println((long)remoteAddressType);
    // write name
    if ( name != NULL ) {
        dc.indent();
        dc.getPs().println("name") ;
        retVal +=name->dump(dc);
    }
    // write ipV4Address
    if ( ipV4Address != NULL ) {
        dc.indent();
        dc.getPs().println("ipV4Address") ;
        retVal +=ipV4Address->dump(dc);
    }
    // write ipV6Address
    if ( ipV6Address != NULL ) {
        dc.indent();
        dc.getPs().println("ipV6Address") ;
        retVal +=ipV6Address->dump(dc);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for protocolFlag
//ui8 ServiceSocket::getProtocolFlag()
//{
//return this->protocolFlag;
//}

// Setter for protocolFlag
//void ServiceSocket::setProtocolFlag(ui8 val)
//{
//this->protocolFlag= val;
//}

// Getter for localPort
//ui16 ServiceSocket::getLocalPort()
//{
//return this->localPort;
//}

// Setter for localPort
//void ServiceSocket::setLocalPort(ui16 val)
//{
//this->localPort= val;
//}

// Getter for remotePort
//ui16 ServiceSocket::getRemotePort()
//{
//return this->remotePort;
//}

// Setter for remotePort
//void ServiceSocket::setRemotePort(ui16 val)
//{
//this->remotePort= val;
//}

// Getter for remoteAddressType
//ui8 ServiceSocket::getRemoteAddressType()
//{
//return this->remoteAddressType;
//}

// Setter for remoteAddressType
//void ServiceSocket::setRemoteAddressType(ui8 val)
//{
//this->remoteAddressType= val;
//}

// Getter for name
//org_himalay_ccif_ByteArrayWrapper_ui8__PTR_TYPE ServiceSocket::getName()
//{
//return this->name;
//}

// Setter for name
//void ServiceSocket::setName(org_himalay_ccif_ByteArrayWrapper_ui8__PTR_TYPE val)
//{
//this->name= val;
//}

// Getter for ipV4Address
//org_himalay_ccif_IPV4Address__PTR_TYPE ServiceSocket::getIpV4Address()
//{
//return this->ipV4Address;
//}

// Setter for ipV4Address
//void ServiceSocket::setIpV4Address(org_himalay_ccif_IPV4Address__PTR_TYPE val)
//{
//this->ipV4Address= val;
//}

// Getter for ipV6Address
//org_himalay_ccif_IPV6Address__PTR_TYPE ServiceSocket::getIpV6Address()
//{
//return this->ipV6Address;
//}

// Setter for ipV6Address
//void ServiceSocket::setIpV6Address(org_himalay_ccif_IPV6Address__PTR_TYPE val)
//{
//this->ipV6Address= val;
//}

int ServiceSocket::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_ServiceSocket__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::ServiceSocket* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::ServiceSocket* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code